Code/Editor/Core/QtEditorApplication.cpp

```diff
@@ -456,55 +437,16 @@ namespace Editor
 
     void EditorQtApplication::EnableOnIdle(bool enable)
     {
+        m_applicationActive = enable;
         if (enable)
         {
-            if (m_idleTimer->interval() == UninitializedFrequency)
-            {
-                ResetIdleTimerInterval();
-            }
-
-            m_idleTimer->start();
-        }
-        else
-        {
-            m_idleTimer->stop();
+            QTimer::singleShot(0, this, &EditorQtApplication::maybeProcessIdle);
         }
     }
 
     bool EditorQtApplication::OnIdleEnabled() const
```

Comment:
OnIdleEnabled is a weird name for a getter function

Reply:
Hah, I thought that too. But it tests whether the "OnIdle" processing is "Enabled" or not. I didn't see the need to change the function names to fix this bug.